### PR TITLE
Convert codegen, metrics and proto to Maven submodules

### PR DESCRIPTION
**Pull request by Marcono1234**
Converts the `codegen`, `metrics` and `proto` projects to Maven submodules.
This does not necessary add much value for these projects (see #2007, and note that `metrics` has to be run manually), but it at least cleans up these projects and makes sure they compile / build.

I have marked this pull request as draft for now; please let me know what you think.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -65,35 +68,6 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
-          <executions>
```
Comment (Marcono1234) [inline pom.xml:68]: Have moved this to the `gson` submodule.

File: metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java [hunk 2]
```diff
@@ -16,14 +16,6 @@
 
 package com.google.gson.metrics;
 
-import com.google.caliper.Param;
```
Comment (eamonnmcmanus) [inline metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java:19]: Google's style rules concerning imports have changed over time, but I think this file follows the current guidelines now and doesn't after these changes. The imports should be in lexicographical order.

**Review by eamonnmcmanus — CHANGES_REQUESTED** [review]
I think this makes sense. As discussed in #2007, we may just delete the `codegen` project.

File: metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java (hunk 2 above)
Comment (Marcono1234) [reply metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java:19]: Sorry, have fixed this now. I was using Eclipse's save actions and had imported the Google Style Guide formatter settings, but had not adjusted the additional Eclipse settings as described in https://github.com/google/styleguide/issues/273#issuecomment-457843194.

Would it make sense to add the [spotless Maven plugin](https://github.com/diffplug/spotless/tree/main/plugin-maven#google-java-format) to enforce the style guide, or was your concern here mainly that the formatting of an existing file was changed?

Comment (eamonnmcmanus) [reply metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java:19]: My main concern about doing that would be that we would have to reformat everything to conform, which would invalidate most outstanding PRs. (I have the same concern about updating the tests to JUnit4 + Truth.)

**Review by eamonnmcmanus — APPROVED** [review]

File: metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java (hunk 2 above)
Comment (Marcono1234) [reply metrics/src/main/java/com/google/gson/metrics/ParseBenchmark.java:19]: Ok, might be best to not apply such a plugin then at the moment.

**Pull request merged**